### ISSUE TITLE
be confused about sequence_generator.py

### ISSUE THREAD

**Issue by travel-go** [pytorch/fairseq#173]
hello, I am confused about the attn_buff in sequence_generator.py,I found it not used anywhere

**Comment by travel-go** [comment]
I would like to know what is the effect of the attn and attn_buff in sequence_generator.py?

**Comment by myleott** [comment]
It just keeps track of the attention mask during generation, so that we can output the alignment. `attn_buf` is swapped with `attn` here: https://github.com/fairinternal/fairseq-py/blob/3b987a818e52cc9b86e534e8f3acea39f55aa82c/fairseq/sequence_generator.py#L489-L490

**Comment by travel-go** [comment]
oh,Thank you!

**Comment by BitMindLab** [comment]
what's the purpuse of attn_buf
